### PR TITLE
BREAKING: stop embedding 'micrometer-core' and 'micrometer-jvm-extras'

### LINKED ISSUE #96: AEE multiple versions of Micrometer on the classpath can cause troubles
[Relates to #20]

AS-IS:
- Alfred Telemetry ships with e specific version of Micrometer (1.0.6 at the moment of this writing)
- Alfresco **Enterprise** Edition >= 6.1 includes a version of Micrometer as well

If there are incompatible changes between those two Micrometer versions, this might cause issues. 

* A specific example:  when starting AEE 6.2.2 with Alfred Telemetry 0.5.0 installed, we observer following error:

    ```
    Caused by: java.lang.NoSuchMethodError: 'io.micrometer.core.instrument.MeterRegistry$Config 
    io.micrometer.core.instrument.MeterRegistry$Config.onMeterRemoved(java.util.function.Consumer)'
    ```

    In this particular case, the out of box Alfresco metrics fail to start because of an incompatibility with the Micrometer version 
    that is actually used in the classpath. In this particular example we can disable Alfresco out of the box metrics as a 
    workaround (`metrics.enabled=false`). Please note that Alfred Telemetry metrics will still be initialized and available. 

* Another example is #20. 
    In that case, Alfred Telemetry failed to start because we were using Micrometer API that was removed in the version shipped with Alfresco. We worked around the problem by refactoring / removing usage of the missing Micrometer API.

To conclude, the main problem here is that the behavior of the out of box Alfresco (enterprise) metrics and Alfred Telemetry is nondeterministic since it might depend on the Micrometer version that is actually used by the classloader. As long as both Alfresco and Alfred Telemetry put different versions on the classpath, there might be incompatibilities and problems in either Alfresco OOTB metrics or Alfred Telemetry metrics.

### PR DESCRIPTION
**Pull request by tgeens**
On ACE (all versions) or EE < v6.1, you need to add the micrometer dependencies yourself:

```groovy
alfrescoSM "io.micrometer:micrometer-core:${micrometerVersion}"
alfrescoSM "io.github.mweirauch:micrometer-jvm-extras:${jvmExtrasVersion}"
```

Alfresco will crash at startup if `alfred-telemetry` is installed, but the required dependencies are missing, with following errors respectively:

`micrometer-core` missing:
```
java.lang.RuntimeException: Required dependency 'io.micrometer:micrometer-core' is not found on the classpath!
        at eu.xenit.alfred.telemetry.service.MicrometerOnClasspathDetector.postProcessBeanDefinitionRegistry(MicrometerOnClasspathDetector.java:14)
```

`micrometer-jvm-extras` missing:
```
Caused by: java.lang.RuntimeException: Missing dependency 'io.github.mweirauch:micrometer-jvm-extras' or disable binder with 'alfred.telemetry.binder.process.enabled=false'
        at eu.xenit.alfred.telemetry.binder.ProcessMetrics.bindTo(ProcessMetrics.java:20)

```

This fixes #96

### CODE REVIEW

File: integration-tests/alfresco-enterprise-62/overload.gradle [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ext {
-    alfrescoBaseWar = 'org.alfresco:content-services:6.2.1@war'
-    alfrescoBaseImage = 'hub.xenit.eu/alfresco-enterprise/alfresco-repository-enterprise:6.2.1'
+    alfrescoBaseWarBom = 'org.alfresco:acs-packaging:6.2.2.19'
+    alfrescoBaseWar = 'org.alfresco:content-services@war'
+    alfrescoBaseImage = 'hub.xenit.eu/alfresco-enterprise/alfresco-repository-enterprise:6.2.2'
```
Comment (vierbergenlars) [inline integration-tests/alfresco-enterprise-62/overload.gradle:4]: Can we pin this base image to exactly `6.2.2.19`?

Else, integration tests will fail when docker-alfresco builds a new image for a future Alfresco version.

**Comment by hvancamp** [comment]
@tgeens can this PR be picked up again please?

File: build.gradle [hunk 2]
```diff
@@ -37,6 +37,7 @@ allprojects {
         hamcrestVersion = '1.3'
         awaitilityVersion = '4.1.0'
         restAssuredVersion = '4.0.0'
+        assertjVersion = '3.16.1'
```
[the remark this answers is not in the document]
Comment (tgeens) [reply build.gradle]: Correct, it was added in a failed attempt to fix test-coverage using classpath magic

File: integration-tests/alfresco-enterprise-62/overload.gradle (hunk 1 above)
Comment (tgeens) [reply integration-tests/alfresco-enterprise-62/overload.gradle]: Pinned

**Review by vierbergenlars — APPROVED** [review]

**Pull request merged**